desc/compute/_profiles.py

```diff
@@ -572,279 +540,696 @@ def _iota_r(params, transforms, profiles, data, **kwargs):
+        - beta * data["sqrt(g)_r"]
     ) / data["sqrt(g)"]
-    num_rr = (
+    beta_rr = (
         data["lambda_rrz"] * data["g_tt"]
         + 2 * data["lambda_rz"] * data["g_tt_r"]
         + data["lambda_z"] * data["g_tt_rr"]
         - data["lambda_rrt"] * data["g_tz"]
         - 2 * data["lambda_rt"] * data["g_tz_r"]
         - (1 + data["lambda_t"]) * data["g_tz_rr"]
-        - 2 * num_r * data["sqrt(g)_r"]
-        - num * data["sqrt(g)_rr"]
+        - 2 * beta_r * data["sqrt(g)_r"]
+        - beta * data["sqrt(g)_rr"]
     ) / data["sqrt(g)"]
-    data["iota_zero_current_num_rr"] = surface_averages(transforms["grid"], num_rr)
-    # TODO: limit at axis
+    beta_rrr = transforms["grid"].replace_at_axis(
+        (
+            data["lambda_rrrz"] * data["g_tt"]
+            + 3 * data["lambda_rrz"] * data["g_tt_r"]
+            + 3 * data["lambda_rz"] * data["g_tt_rr"]
+            + data["lambda_z"] * data["g_tt_rrr"]
+            - data["lambda_rrrt"] * data["g_tz"]
+            - 3 * data["lambda_rrt"] * data["g_tz_r"]
+            - 3 * data["lambda_rt"] * data["g_tz_rr"]
+            - (1 + data["lambda_t"]) * data["g_tz_rrr"]
+            - 3 * beta_rr * data["sqrt(g)_r"]
+            - 3 * beta_r * data["sqrt(g)_rr"]
+            - beta * data["sqrt(g)_rrr"]
+        )
+        / data["sqrt(g)"],
+        # Todo: axis limit of beta_rrr
+        #   Computed with four applications of l’Hôpital’s rule.
+        #   Requires sqrt(g)_rrrr and fourth derivatives of basis vectors.
+        jnp.nan,
+    )
+    beta_rrr = surface_integrals(transforms["grid"], beta_rrr)
+    # force limit to nan until completed because integration replaces nan with 0
+    data["iota_num_rrr"] = alpha_rrr + transforms["grid"].replace_at_axis(
+        beta_rrr, jnp.nan
+    )
     return data
 
 
 @register_compute_fun(
-    name="iota_zero_current_den",
-    label="\\iota_{0~\\mathrm{denominator}}",
+    name="iota_den",
+    label="\\iota_{\\mathrm{denominator}}",
     units="m^{-1}",
     units_long="inverse meters",
-    description="Zero toroidal current rotational transform denominator",
+    description="Denominator of rotational transform formula",
     dim=1,
     params=[],
     transforms={"grid": []},
-    profiles=[],
+    profiles=["current"],
     coordinates="r",
-    data=["g_tt", "sqrt(g)"],
-    axis_limit_data=["g_tt_rr", "sqrt(g)_r"],
+    data=["0", "g_tt", "g_tz", "sqrt(g)", "omega_t", "omega_z"],
 )
-def _iota_zero_current_den(params, transforms, profiles, data, **kwargs):
-    den = data["g_tt"] / data["sqrt(g)"]
-    data["iota_zero_current_den"] = surface_averages(transforms["grid"], den)
+def _iota_den(params, transforms, profiles, data, **kwargs):
+    """Denominator of rotational transform formula.
+
+    Computes 𝛾 as defined in the document attached to the description
+    of GitHub pull request #556.
+    """
+    if profiles["current"] is None:
+        data["iota_den"] = jnp.nan * data["0"]
+        return data
```

Comment:
These could be computed just fine on fixed iota equilibrium. I placed this guard to avoid unneded computation. However the all the dependencies needed to compute `iota_den` etc. are computed even on fixed iota equilibriums that don't need this computation.